extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -181,14 +158,23 @@ public void getSwaggerUiFinalDestination(
     public void registerSwaggerUiHandler(SwaggerUiRecorder recorder,
             BuildProducer<RouteBuildItem> routes,
             NonApplicationRootPathBuildItem nonApplicationRootPathBuildItem,
-            SwaggerUiBuildItem finalDestinationBuildItem,
+            WebJarResultBuildItem webJarResultBuildItem,
             SwaggerUiRuntimeConfig runtimeConfig,
             LaunchModeBuildItem launchMode,
-            SwaggerUiConfig swaggerUiConfig) throws Exception {
+            SwaggerUiConfig swaggerUiConfig,
+            BuildProducer<SwaggerUiBuildItem> swaggerUiBuildProducer) {
+
+        WebJarResultBuildItem.WebJarResult result = webJarResultBuildItem.byArtifactKey(SWAGGER_UI_WEBJAR_ARTIFACT_KEY);
+        if (result == null) {
+            return;
+        }
 
         if (shouldInclude(launchMode, swaggerUiConfig)) {
-            Handler<RoutingContext> handler = recorder.handler(finalDestinationBuildItem.getSwaggerUiFinalDestination(),
-                    finalDestinationBuildItem.getSwaggerUiPath(),
+            String swaggerUiPath = nonApplicationRootPathBuildItem.resolvePath(swaggerUiConfig.path);
+            swaggerUiBuildProducer.produce(new SwaggerUiBuildItem(result.getFinalDestination(), swaggerUiPath));
```

Comment:
`SwaggerUiBuildItem` just to be on the safe side, no idea if anyone consumes it.

Maybe we can deprecate it also?